### ISSUE TITLE
Executable named git not found on path

### ISSUE THREAD

**Issue by jmitchell** [input-output-hk/stack2nix#11]
On a nixos machine I'm seeing that having the following in `~/.stack/config.yaml` causes a runtime failure.

```
nix:
  enable: true
```

From outside a nix-shell:

```
$ stack2nix -o /tmp/haskell-multi-package-demo1.nix --revision e3d9bd6d6066dab5222ce53fb7d234f28eafa2d5 https://github.com/jmitchell/haskell-multi-package-demo1.git
Updating deps from /tmp/s2n-25091
stack2nix: FAILED: stack list-dependencies
Command failed: cd "/tmp/s2n-25091" && stack list-dependencies --system-ghc --separator -
Executable named git not found on path: ["/home/jake/.stack/snapshots/x86_64-linux-nix/lts-8.9/8.0.2/bin","/nix/store/dvlcd6yb874y5y1jkw99mxrrpwh7dn6a-ghc-8.0.2/bin","/nix/store/29cj9f40783rkqzzmp2n70rbzfvnmykk-patchelf-0.9/bin","/nix/store/b28rci2pd2bjrcgj76x594phflbd6sv8-paxctl-0.9/bin","/nix/store/k8dyhz9hrjydl23ny3131am6cws1rc07-coreutils-8.26/bin","/nix/store/dqzqrrampaf665laggc4ng33dfbxv5q7-findutils-4.6.0/bin","/nix/store/2arb589k64xmjfw3aa819dz5iqxiyqrd-diffutils-3.5/bin","/nix/store/2nzkvn5b9jxdj75mgw6cdvpl6438sdqw-gnused-4.4/bin","/nix/store/3r21p23hmbzj1ykv2wllqr14k1b0cdmx-gnugrep-3.0/bin","/nix/store/8hwvdkr7i3gbkyp50lq5dpgb7fvniryv-gawk-4.1.3/bin","/nix/store/255w01kim6i6h604av7fxh63lplbldgb-gnutar-1.29/bin","/nix/store/m1z4gaqwal3gkhjz8309n2c24slr3dma-gzip-1.8/bin","/nix/store/6fc4ak60p1mnilrjfcyz0sywwmpmf45w-bzip2-1.0.6.0.1-bin/bin","/nix/store/bnjmyzdv6l0aw4xw4z2fr59qv75lbz88-gnumake-4.2.1/bin","/nix/store/fi3mbd2ml4pbgzyasrlnp0wyy6qi48fh-bash-4.4-p5/bin","/nix/store/6ygw78s2wpw8smvdrig7y05m2z1na4md-patch-2.7.5/bin","/nix/store/s2hmf51bprsxb20v3bfigs8a7zsw3hjl-xz-5.2.2-bin/bin"]


CallStack (from HasCallStack):
  error, called at src/Stack2nix.hs:195:25 in stack2nix-0.1.0.0-5CQNnxswc3zCt1pRgRw9R5:Stack2nix
```

The cited line corresponds to the `stack list-dependencies --system-ghc` call. Adding an explicit `--nix` to the call doesn't seem to help, and neither does removing `--system-ghc`. The only thing that seems to help is removing the global stack configuration setting enabling nix.

Running the following manually worked just fine, even with nix integration globally enabled. It seems PATH rewriting is happening only when `stack` is invoked from `stack2nix`.

```
$ git clone https://github.com/jmitchell/haskell-multi-package-demo1.git tmp
$ cd tmp
$ git checkout e3d9bd6d6066dab5222ce53fb7d234f28eafa2d5
$ stack list-dependencies --system-ghc --separator -
```

Happy to have narrowed in on the issue a little. It's happened sporadically, on various test machines and I've only now pinned it to the nix global integration setting in ~/.stack/config.yaml.

**Comment by domenkozar** [comment]
Seems like something is using pure nix-shell resulting into `git` not being available.

**Comment by domenkozar** [comment]
I've managed to reproduce.

Short term fix is 

```
diff --git a/stack.yaml b/stack.yaml
index c623962..ccac384 100644
--- a/stack.yaml
+++ b/stack.yaml
@@ -13,4 +13,7 @@ extra-deps: []

 flags: {}

+nix:
+  packages: [git]
+
 extra-package-dbs: []
```

Long term fix would be for stack to include git inside the shell

**Comment by domenkozar** [comment]
Relevant upstream issue: https://github.com/commercialhaskell/stack/issues/2089

**Comment by domenkozar** [comment]
Upstream PR: https://github.com/commercialhaskell/stack/pull/3205

**Comment by domenkozar** [comment]
Upstream has merged the PR, meanwhile you can use the workaround :)

**Comment by domenkozar** [comment]
This has been released upstream in Stack 1.5, so I'm closing this.
